src/python/gudhi/representations/vector_methods.py

```diff
@@ -350,6 +351,92 @@ def __call__(self, diag):
+        See the corresponding function in BettiCurve.
+        """
+        if np.isnan(self.filtration_range).any():
+            pre = DiagramScaler(use=True, scalers=[([0], MinMaxScaler()), ([1], MinMaxScaler())]).fit(X,y)
+            [mx,my],[Mx,My] = [pre.scalers[0][1].data_min_[0], pre.scalers[1][1].data_min_[0]], [pre.scalers[0][1].data_max_[0], pre.scalers[1][1].data_max_[0]]
+            self.filtration_range = np.where(np.isnan(self.filtration_range), np.array([mx, My]), np.array(self.filtration_range))
+
+        return self
+
+    def transform(self, X):
+        """
+        Compute the Betti curve for each persistence diagram.
+
+        Parameters:
+            X (list of n x 2 NumPy arrays): input persistence diagrams.
+        
+        Returns:
+            Pair (xs, bettis), where xs is a length-k NumPy array with all filtration values for which any of the Betti curves change value, and bettis is an (number of diagrams) x k NumPy integer array with the values of the k Betti curves at the corresponding filtration values. The value bettis[i, k] is the value of the Betti curve of the k'th diagram on the interval [xs[i], xs[i+1]).
+        """
+        assert(not np.isnan(self.filtration_range).any())
+
+        N = len(X)
+        cropped_pds = [pd[pd[:, 0] >= self.filtration_range[0]] for pd in X]
+
+        events = np.concatenate([pd.flatten(order="F") for pd in cropped_pds], axis=0)
+        sorting = np.argsort(events)
+        offsets = np.zeros(1 + N, dtype=int)
+        for i in range(0, N):
+            offsets[i+1] = offsets[i] + 2*cropped_pds[i].shape[0]
+        starts = offsets[0:N]
+        ends = offsets[1:N + 1] - 1
+
+        xs = [self.filtration_range[0]]
+        bettis = [[0] for i in range(0, N)]
+
+        for i in sorting:
+            if events[i] >= self.filtration_range[1]:
+                break
+            j = np.searchsorted(ends, i)
+            delta = 1 if i - starts[j] < len(cropped_pds[j]) else -1
+            if events[i] == xs[-1]:
+                bettis[j][-1] += delta
+            else:
+                xs.append(events[i])
+                for k in range(0, j):
+                    bettis[k].append(bettis[k][-1])
+                bettis[j].append(bettis[j][-1] + delta)
+                for k in range(j+1, N):
+                    bettis[k].append(bettis[k][-1])
+
+        xs = np.array(xs)
+        bettis = np.array(bettis, dtype=int)
+
+        return(xs, bettis)
+
+    def __call__(self, diag):
+        """
+        Apply IrregularBettiCurve on a single persistence diagram and outputs the result.
+
+        Parameters:
+            diag (n x 2 numpy array): input persistence diagram.
+
+        Returns:
+            Pair (xs, b) of 1D NumPy arrays, xs containing the filtration values and b containing the Betti curve values. The Betti curve takes the value b[i] on the interval [xs[i], xs[i+1]).
+        """
+        res = self.fit_transform([diag])
+        return (res[0], res[1][0, :])
```

Comment:
I wonder if we should leave the `xs` as an attribute of the class and just output `bettis`. I am saying this because the class nicely fits into scikit-learn framework, but using it in an sklearn Pipeline will probably fail due to the fact that there are two outputs (I might be wrong though).

Reply:
Sure thing! I'll make a note and make that change.